src/test/scala/chiselTests/naming/PrefixSpec.scala

```diff
@@ -460,17 +460,15 @@ class PrefixSpec extends ChiselPropSpec with Utils {
 
       {
         val x5 = {
-          val x1 = chisel3.assert(1.U === 1.U)
           val x2 = cover(foo =/= bar)
           val x3 = chisel3.assume(foo =/= 123.U)
           val x4 = printf("foo = %d\n", foo)
-          x1
+          x2
         }
       }
     }
     val chirrtl = ChiselStage.emitCHIRRTL(new Test)
-    (chirrtl should include).regex("assert.*: x5")
-    (chirrtl should include).regex("cover.*: x5_x2")
+    (chirrtl should include).regex("cover.*: x5")
     (chirrtl should include).regex("assume.*: x5_x3")
```

Comment:
I know this PR is just a re-application of 2 previously reverted PRs so I'm not going to block on this issue (since it was in a PR I previously approved), but this is not ideal. We should be able to maintain the old label behavior, but it will probably require a new Printable that wraps the Assert() object that we return from `chisel3.assert`.

Reply:
Ah, so instead of "chisel3_builtin" being the label for all of these, they get the "valname"/name included that would previously be mixed in by putting it in the "optional name" (trailing ": asdf" this test checks for)?

Yes that seems rather useful!

So Printable would help get a friendly name available (that's not available when IntrinsicExpr apply is called) so we could append it to the label StringParam while converting to the FIRRTL IR object, thereabouts? Makes sense to me!

Reply:
> So Printable would help get a friendly name available (that's not available when IntrinsicExpr apply is called) so we could append it to the label StringParam while converting to the FIRRTL IR object, thereabouts? Makes sense to me!

Exactly!

> FYI previous behavior was just dropping optional names (assert_ label in the test) emitted for assert op. chisel3_builtin was used regardless of optional names.

Perhaps for these asserts in firtool, but the intent was always to try to replace printf-encoded with real asserts with reasonable label behavior. This has been used by ChiselTest users, we just never fully implemented it in firtool, nor for printf-encoded asserts, but we should.